test/powershell/engine/Job/Jobs.Tests.ps1

```diff
@@ -1,38 +1,287 @@
+            )
+        }
+
+        AfterEach {
+            Get-Job | Where-Object { $_.Id -ne $startedJob.Id } | Remove-Job -ErrorAction SilentlyContinue -Force
+        }
+
+        It 'Can Get-Job with <property>' -TestCases $getJobTestCases {
+            param($parameters)
+            $job = Get-Job @parameters
+            ValidateJobInfo -job $job -state 'Completed' -hasMoreData $true -command ' 1 + 1 '
+        }
+
+        It 'Can Get-Job with <property>' -TestCases $getJobSwitches {
+            param($parameters)
+            $job = Get-Job @parameters
+            ValidateJobInfo -job $job -state 'Completed' -hasMoreData $true -Name 'StartedJob'
+        }
+
+        It 'Can Get-Job with <property>' -TestCases $getJobChildJobs {
+            param($parameters)
+            $jobs = Get-Job @parameters
+            $jobs.Count | Should Be 2
+            ValidateJobInfo -job $jobs[0] -state 'Completed' -hasMoreData $true -Name 'StartedJob'
+            ValidateJobInfo -job $jobs[1] -state 'Completed' -hasMoreData $true
+        }
+    }
+
+
+    Context 'Remove-Job tests' {
+        # The test pattern used here is different from other tests since there is a scoping issue in Pester.
+        # If BeforeEach is used then $removeJobTestCases does not bind when the It is called.
+        # This implementation works around the problem by using a BeforeAll and creating a job inside the It.
+        BeforeAll {
+            $removeJobTestCases = @(
+                @{ property = 'Name'}
+                @{ property = 'Id'}
+                @{ property = 'InstanceId'}
+                @{ property = 'State'}
+            )
+        }
+
+        It 'Can Reemove-Job with <property>' -TestCases $removeJobTestCases {
+            param($property)
+            $jobToRemove = Start-Job -ScriptBlock { 1 + 1 } -Name 'JobToRemove' | Wait-Job
+            $splat = @{ $property = $jobToRemove.$property }
+            Remove-Job @splat
+            Get-Job $jobToRemove -ErrorAction SilentlyContinue | Should BeNullOrEmpty
+        }
+    }
+
+    Context 'Stop-Job tests' {
+        # The test pattern used here is different from other tests since there is a scoping issue in Pester.
+        # If BeforeEach is used then $stopJobTestCases does not bind when the It is called.
+        # This implementation works around the problem by using a BeforeAll and creating a job inside the It.
+        BeforeAll {
+            $stopJobTestCases = @(
+                @{ property = 'Name'}
+                @{ property = 'Id'}
+                @{ property = 'InstanceId'}
+                @{ property = 'State'}
+            )
+
+            $jobToStop = Start-Job -ScriptBlock { Start-Sleep -Seconds 100 } -Name 'JobToStop'
+        }
+
+        It 'Can Stop-Job with <property>' -TestCases $stopJobTestCases {
+            param($property)
+            # '-Seconds 100' is chosen to be substantially large, so that the job is in running state when Stop-Job is called.
```

Comment:
Can you move this comment to line 271? It makes more sense there

Reply:
Moved.